corehq/apps/userreports/views.py

```diff
@@ -73,7 +73,7 @@
 def get_datasource_config_or_404(config_id, domain):
     try:
         return get_datasource_config(config_id, domain)
-    except DataSourceConfigurationNotFoundError:
+    except (DataSourceConfigurationNotFoundError, BadSpecError):
         raise Http404
```

Comment:
I agree that `DataSourceConfigurationNotFoundError` seems more appropriate here, but that's actually being caught [here](https://github.com/dimagi/commcare-hq/blob/16b1a7181f78a93e9f07a5072c08635ee53a7bf2/corehq/apps/userreports/models.py#L267-L273) and turned into a `BadSpecError`, so it feels weird to change it back.  Maybe that *shouldn't* raise a `BadSpecError`?  I don't feel strongly enough about any of these to make a change though.